ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/PDELabelProvider.java

```diff
@@ -203,6 +210,13 @@ public String getObjectText(IPluginBase pluginBase) {
 		}
 		if (pluginBase.getModel() != null && !pluginBase.getModel().isInSync())
 			text += " " + PDEUIMessages.PluginModelManager_outOfSync; //$NON-NLS-1$
+
+		HashMap<String, Boolean> existingImports = PluginSelectionDialog.getExistingImports(currentModel, false);
```

Comment:
Instead of setting `IPluginModelBase currentModel` via a setter and saving it, could you just use the following?
```
if (pluginBase.getModel() instanceof IPluginModelBase pluginModel) {
  ...
```
Because the less state we have the better. I also don't know if `currentModel` can be null at this point in any usage scenario and if the called `PluginSelectionDialog.getExistingImports` method would handle that well.